False Negative: SimplifyBooleanExpression does not report with paranthesized boolean literals in ternary operators

I have read check documentation: https://checkstyle.org/checks/xxxxxx/nameofaffectedcheck.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java 
/var/tmp $ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="SimplifyBooleanExpression"/>
    </module>
</module>
/var/tmp $ cat Test.java 
public class Test {
    public void test(boolean x) {
        if (x && true) {}
        if (x && (true)) {}
        if (x ? true : false) {}
        if (x ? (true) : false) {}
    }
}
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"                                         
/var/tmp $ java $RUN_LOCALE -jar checkstyle-13.4.0-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /mnt/d/IntellijProjects/checkstyle-playground/inputs/simplify-boolean-expression/Test.java:3:15: Expression can be simplified. [SimplifyBooleanExpression]
[ERROR] /mnt/d/IntellijProjects/checkstyle-playground/inputs/simplify-boolean-expression/Test.java:4:15: Expression can be simplified. [SimplifyBooleanExpression]
[ERROR] /mnt/d/IntellijProjects/checkstyle-playground/inputs/simplify-boolean-expression/Test.java:5:15: Expression can be simplified. [SimplifyBooleanExpression]
Audit done.
Checkstyle ends with 3 errors.
```
---

**Describe what you expect in detail.**

All four `if` statements should produce violations, instead of only the first three.

Since paranthesizing `true` does not suppress violation in boolean expressions, it should also not in ternary operators by the same logic.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanExpressionCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanExpressionCheck.java
@@ -75,11 +75,12 @@ public class SimplifyBooleanExpressionCheck
                  TokenTypes.LAND -> log(parent, MSG_KEY);
 
             case TokenTypes.QUESTION -> {
-                final DetailAST nextSibling = ast.getNextSibling();
-                if (TokenUtil.isBooleanLiteralType(parent.getFirstChild().getType())
-                        || nextSibling != null && nextSibling.getNextSibling() != null
+                final DetailAST firstChild = skipParentheses(parent.getFirstChild());
+                final DetailAST nextSibling = skipParentheses(ast.getNextSibling());
+                if (TokenUtil.isBooleanLiteralType(firstChild.getType())
+                        || nextSibling != null
                         && TokenUtil.isBooleanLiteralType(
-                        nextSibling.getNextSibling().getType())) {
+                                skipParentheses(nextSibling.getNextSibling()).getType())) {
                     log(parent, MSG_KEY);
                 }
             }
@@ -88,6 +89,21 @@ public class SimplifyBooleanExpressionCheck
                 // do nothing
             }
         }
+    }
+
+    /**
+     * Iterates sibling nodes, skipping parentheses.
+     *
+     * @param node The starting node.
+     * @return The first sibling not of type {@code TokenTypes.LPAREN} or
+     *     {@code TokenTypes.RPAREN}, or {@code null} if no such node exists.
+     */
+    private static DetailAST skipParentheses(DetailAST node) {
+        DetailAST result = node;
+        while (TokenUtil.isOfType(result, TokenTypes.LPAREN, TokenTypes.RPAREN)) {
+            result = result.getNextSibling();
+        }
+        return result;
     }
 
 }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanExpressionCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanExpressionCheckTest.java
@@ -37,19 +37,23 @@ public class SimplifyBooleanExpressionCheckTest
     @Test
     public void testIt() throws Exception {
         final String[] expected = {
-            "22:18: " + getCheckMessage(MSG_KEY),
-            "43:36: " + getCheckMessage(MSG_KEY),
-            "44:36: " + getCheckMessage(MSG_KEY),
-            "45:16: " + getCheckMessage(MSG_KEY),
-            "45:32: " + getCheckMessage(MSG_KEY),
-            "95:27: " + getCheckMessage(MSG_KEY),
-            "96:24: " + getCheckMessage(MSG_KEY),
-            "98:27: " + getCheckMessage(MSG_KEY),
-            "104:23: " + getCheckMessage(MSG_KEY),
-            "106:17: " + getCheckMessage(MSG_KEY),
-            "109:21: " + getCheckMessage(MSG_KEY),
-            "110:23: " + getCheckMessage(MSG_KEY),
-            "111:20: " + getCheckMessage(MSG_KEY),
+            "20:18: " + getCheckMessage(MSG_KEY),
+            "41:36: " + getCheckMessage(MSG_KEY),
+            "42:36: " + getCheckMessage(MSG_KEY),
+            "43:16: " + getCheckMessage(MSG_KEY),
+            "43:32: " + getCheckMessage(MSG_KEY),
+            "93:27: " + getCheckMessage(MSG_KEY),
+            "94:24: " + getCheckMessage(MSG_KEY),
+            "96:27: " + getCheckMessage(MSG_KEY),
+            "102:23: " + getCheckMessage(MSG_KEY),
+            "104:17: " + getCheckMessage(MSG_KEY),
+            "107:21: " + getCheckMessage(MSG_KEY),
+            "108:23: " + getCheckMessage(MSG_KEY),
+            "109:20: " + getCheckMessage(MSG_KEY),
+            "111:27: " + getCheckMessage(MSG_KEY),
+            "112:35: " + getCheckMessage(MSG_KEY),
+            "114:31: " + getCheckMessage(MSG_KEY),
+            "115:35: " + getCheckMessage(MSG_KEY),
         };
         verifyWithInlineConfigParser(
                 getPath("InputSimplifyBooleanExpression.java"), expected);

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanExpressionCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanExpressionCheckTest.java
@@ -50,6 +50,8 @@ public class SimplifyBooleanExpressionCheckTest
             "109:21: " + getCheckMessage(MSG_KEY),
             "110:23: " + getCheckMessage(MSG_KEY),
             "111:20: " + getCheckMessage(MSG_KEY),
+            "113:27: " + getCheckMessage(MSG_KEY),
+            "114:31: " + getCheckMessage(MSG_KEY),
         };
         verifyWithInlineConfigParser(
                 getPath("InputSimplifyBooleanExpression.java"), expected);

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanexpression/InputSimplifyBooleanExpression.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanexpression/InputSimplifyBooleanExpression.java
@@ -8,12 +8,10 @@ package com.puppycrawl.tools.checkstyle.checks.coding.simplifybooleanexpression;
 
 /**
    Contains boolean logic that can be simplified.
-
    @author lkuehne
  */
 public class InputSimplifyBooleanExpression
 {
-
     public static boolean isOddMillis()
     {
         boolean even = System.currentTimeMillis() % 2 == 0;
@@ -110,5 +108,13 @@ public class InputSimplifyBooleanExpression
         int d = false ? 1 : 2; // violation, 'Expression can be simplified'
         temp = a() ? true : true; // violation, 'Expression can be simplified'
         temp = value != null ? value : (false);
+        boolean w = c > 1 ? (true) : ((false)); // violation, 'Expression can be simplified'
+        boolean wNoParens = c > 1 ? true : false; // violation, 'Expression can be simplified'
+        boolean wFullySimplified = c > 1;
+        boolean x = ((false)) ? c == 1 : c > 10; // violation, 'Expression can be simplified'
+        boolean xNoParens = false ? c == 1 : c > 10; // violation, 'Expression can be simplified'
+        boolean xFullySimplified = c > 10;
+        boolean y = c > 1 ? (c < 10 ? false : a) : false;
+        boolean z = (c < 10 ? false : a) ? c == 1 : c > 10;
     }
 }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanexpression/InputSimplifyBooleanExpression.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanexpression/InputSimplifyBooleanExpression.java
@@ -110,5 +110,9 @@ public class InputSimplifyBooleanExpression
         int d = false ? 1 : 2; // violation, 'Expression can be simplified'
         temp = a() ? true : true; // violation, 'Expression can be simplified'
         temp = value != null ? value : (false);
+        boolean w = c > 1 ? (true) : ((false)); // violation, 'Expression can be simplified'
+        boolean x = ((false)) ? c == 1 : c > 10; // violation, 'Expression can be simplified'
+        boolean y = c > 1 ? (c < 10 ? false : a) : false;
+        boolean z = (c < 10 ? false : a) ? c == 1 : c > 10;
     }
 }

